### PR TITLE
RFC: deduplicate angle to sexagesimal functions in coordinates.angles.formats

### PR DESCRIPTION
**Pull request by neutrinoceros**
### Description
This pull request is to address part of #16082

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
> This file claims it's not a part of public API, so can afford to be quite aggressive.

ow I completely missed that, thank you for pointing it out !

File: astropy/coordinates/angles/formats.py [hunk 1]
```diff
@@ -399,32 +399,32 @@ def parse_angle(angle, unit=None, debug=False):
     return _AngleParser().parse(angle, unit, debug=debug)
 
 
+def _angle_to_sexagesimal(a, /):
```
Comment (eerovaher) [inline astropy/coordinates/angles/formats.py:402]: This might be a better name:
```suggestion
def _decimal_to_sexagesimal(a, /):
```

**Comment by neutrinoceros** [comment]
tests seem to pass locally but I'll switch to draft until I'm sure the more agressive patch doesn't break anything in CI

File: astropy/coordinates/angles/formats.py (hunk 1 above)
Comment (neutrinoceros) [reply astropy/coordinates/angles/formats.py]: thanks, I'll use that in my next push !

**Comment by neutrinoceros** [comment]
A nice side effect of applying @eerovaher 's suggestions is that now, #16082 becomes redundant, and the present PR now closes #16082 !

File: astropy/coordinates/angles/formats.py [hunk 2]
```diff
@@ -399,42 +399,28 @@ def parse_angle(angle, unit=None, debug=False):
     return _AngleParser().parse(angle, unit, debug=debug)
 
 
-def degrees_to_dms(d):
+def decimal_to_sexagesimal(a, /):
```
Comment (eerovaher) [inline astropy/coordinates/angles/formats.py:402]:
```suggestion
def _decimal_to_sexagesimal(a, /):
```

File: astropy/coordinates/angles/formats.py [hunk 3]
```diff
@@ -399,42 +399,28 @@ def parse_angle(angle, unit=None, debug=False):
...
-    (mf, m) = np.modf(df * 60.0)  # (minute fraction, minute)
+    # assuming a in degree, these are (arcminute fraction, arcminute)
+    (mf, m) = np.modf(df * 60.0)
     s = mf * 60.0
 
     return np.floor(sign * d), sign * np.floor(m), sign * s
 
 
-def hours_to_hms(h):
+def decimal_to_string(angle, precision=None, pad=False, sep=(":",), fields=3):
```
Comment (eerovaher) [inline astropy/coordinates/angles/formats.py:419]:
```suggestion
def _decimal_to_sexagesimal_string(angle, precision=None, pad=False, sep=(":",), fields=3):
```

**Review by eerovaher — COMMENTED** [review]
It would be best to start the function names with an underscore, so that it would be obvious that they are private.

File: astropy/coordinates/angles/formats.py (hunk 2 above)
Comment (neutrinoceros) [reply astropy/coordinates/angles/formats.py]: done ! (I did it manually so I could change the definitions and call sites in one go)

File: astropy/coordinates/angles/formats.py (hunk 3 above)
Comment (neutrinoceros) [reply astropy/coordinates/angles/formats.py]: done !

Comment (eerovaher) [reply astropy/coordinates/angles/formats.py]: I didn't point this out in any text, but the name of this function should still include "sexagesimal" somewhere.

Comment (neutrinoceros) [reply astropy/coordinates/angles/formats.py]: sorry I missed it ! done

**Review by eerovaher — APPROVED** [review]
I'm not merging quite yet. Maybe @mhvk wants to have a look at this too, and even if he doesn't `milestone_checker` wouldn't let me merge anyways.

**Pull request merged**